golem-worker-executor-base/tests/compatibility/v1.rs

```diff
@@ -0,0 +1,1890 @@
+    SerializableResponse, SerializableResponseHeaders, SerializableTlsAlertReceivedPayload,
+};
+use golem_worker_executor_base::durable_host::serialized::{
+    SerializableDateTime, SerializableError, SerializableFileTimes, SerializableIpAddress,
+    SerializableIpAddresses, SerializableStreamError,
+};
+use golem_worker_executor_base::durable_host::wasm_rpc::serialized::SerializableInvokeResult;
+use golem_worker_executor_base::error::GolemError;
+use golem_worker_executor_base::model::InterruptKind;
+use golem_worker_executor_base::services::blob_store;
+use golem_worker_executor_base::services::promise::RedisPromiseState;
+use golem_worker_executor_base::services::rpc::RpcError;
+use golem_worker_executor_base::services::worker_proxy::WorkerProxyError;
+use std::collections::HashMap;
+use std::fmt::Debug;
+use std::io::Write;
+use std::path::Path;
+use std::time::Duration;
+use uuid::Uuid;
+
+fn is_deserializable<T: Encode + Decode + PartialEq + Debug>(old: &Path, new: &Path) {
+    let old = std::fs::read(old).unwrap();
+    let new = std::fs::read(new).unwrap();
+
+    // Both the old and the latest binary can be deserialized
+    let old_decoded: T = deserialize(&old).unwrap();
+    let new_decoded: T = deserialize(&new).unwrap();
+
+    // And they represent the same value
+    assert_eq!(old_decoded, new_decoded);
+}
+
+fn backward_compatible<T: Encode + Decode + PartialEq + Debug + 'static>(
+    name: impl AsRef<str>,
+    mint: &mut Mint,
+    value: T,
+) {
+    let mut file = mint
+        .new_goldenfile_with_differ(
+            format!("{}.bin", name.as_ref()),
+            Box::new(is_deserializable::<T>),
+        )
+        .unwrap();
+    let encoded = serialize(&value).unwrap();
+    file.write_all(&encoded).unwrap();
+    file.flush().unwrap();
+}
+
+fn is_deserializable_wit_value(old: &Path, new: &Path) {
+    let old = std::fs::read(old).unwrap();
+    let new = std::fs::read(new).unwrap();
+
+    // Both the old and the latest binary can be deserialized
+    let old_decoded: WitValue = deserialize(&old).unwrap();
+    let new_decoded: WitValue = deserialize(&new).unwrap();
+
+    let old_value: Value = old_decoded.into();
+    let new_value: Value = new_decoded.into();
+
+    // And they represent the same value
+    assert_eq!(old_value, new_value);
+}
+
+/// Special case for WitValue which does not implement PartialEq at the moment, but can be converted
+/// to Value for comparison.
+fn backward_compatible_wit_value(name: impl AsRef<str>, mint: &mut Mint, value: WitValue) {
+    let mut file = mint
+        .new_goldenfile_with_differ(
+            format!("{}.bin", name.as_ref()),
+            Box::new(is_deserializable_wit_value),
+        )
+        .unwrap();
+    let encoded = serialize(&value).unwrap();
+    file.write_all(&encoded).unwrap();
+    file.flush().unwrap();
+}
+
+#[test]
+pub fn worker_status() {
+    let ws1 = WorkerStatus::Running;
```

Comment:
are there also tests for not introducing new enum cases?

Reply:
I don't think we need to explicitly deny creating new enum cases. The test covers all the ones that were known for 1.0. So what we are testing is that we have to be able to deserialize all the possible cases created by 1.0. This is covered here. So if you can introduce a new case without breaking this, that's fine (I believe with bincode we can do it if we add the case to the end, but haven't tried).